### ISSUE TITLE
Date picker (iOS) is missing buttons on smaller screens

### ISSUE THREAD

**Issue by westy92** [flutter/flutter#17745]
## Steps to Reproduce

**Note:** Everything works as expected on a larger screen (iPhone X, iOS 11.3)

**Problem device:** Apple 32GB iPod touch (5th generation) running iOS 9.3.5

1. Open date picker widget on iOS
2. OK and CANCEL buttons are missing

![img_0001](https://user-images.githubusercontent.com/290167/40273824-8e406184-5b8d-11e8-92a1-6e54af5a60db.PNG)
![img_0002](https://user-images.githubusercontent.com/290167/40273825-8fcc2178-5b8d-11e8-977e-236dd09b9f71.PNG)
View when debugging:
![img_0004](https://user-images.githubusercontent.com/290167/40273884-b7c9058c-5b8e-11e8-9843-43140e65fe6c.PNG)

## Logs

This is logged when I open the date picker:
```
[+12623 ms] flutter: ══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
[  +24 ms] flutter: The following message was thrown during layout:
[        ] flutter: A RenderFlex overflowed by 38 pixels on the bottom.
[ +122 ms] flutter: 
[   +6 ms] flutter: The overflowing RenderFlex has an orientation of Axis.vertical.
[   +1 ms] flutter: The edge of the RenderFlex that is overflowing has been marked in the rendering with a yellow and
[   +8 ms] flutter: black striped pattern. This is usually caused by the contents being too big for the RenderFlex.
[        ] flutter: Consider applying a flex factor (e.g. using an Expanded widget) to force the children of the
[        ] flutter: RenderFlex to fit within the available space instead of being sized to their natural size.
[        ] flutter: This is considered an error condition because it indicates that there is content that cannot be
[        ] flutter: seen. If the content is legitimately bigger than the available space, consider clipping it with a
[   +2 ms] flutter: ClipRect widget before putting it in the flex, or using a scrollable container rather than a Flex,
[   +1 ms] flutter: like a ListView.
[   +4 ms] flutter: The specific RenderFlex in question is:
[   +7 ms] flutter:   RenderFlex#d2ed7 relayoutBoundary=up7 OVERFLOWING
[        ] flutter:   creator: Column ← SizedBox ← LayoutBuilder ← OrientationBuilder ← DefaultTextStyle ←
[   +2 ms] flutter:   AnimatedDefaultTextStyle ← _InkFeatures-[GlobalKey#39ed3 ink renderer] ←
[        ] flutter:   NotificationListener<LayoutChangedNotification> ← CustomPaint ← _ShapeBorderPaint ← PhysicalShape
[   +1 ms] flutter:   ← _MaterialInterior ← ⋯
[        ] flutter:   parentData: <none> (can use size)
[   +3 ms] flutter:   constraints: BoxConstraints(w=240.0, 0.0<=h<=450.0)
[   +4 ms] flutter:   size: Size(240.0, 450.0)
[   +5 ms] flutter:   direction: vertical
[        ] flutter:   mainAxisAlignment: start
[        ] flutter:   mainAxisSize: min
[        ] flutter:   crossAxisAlignment: stretch
[        ] flutter:   verticalDirection: down
[   +3 ms] flutter: ◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤
[   +1 ms] flutter: ════════════════════════════════════════════════════════════════════════════════════════════════════

```

```
No issues found! (ran in 2.9s)
```

```
[✓] Flutter (Channel dev, v0.4.4, on Mac OS X 10.13.4 17E202, locale en-US)
    • Flutter version 0.4.4 at /Users/seth/Source/flutter
    • Framework revision f9bb4289e9 (8 days ago), 2018-05-11 21:44:54 -0700
    • Engine revision 06afdfe54e
    • Dart version 2.0.0-dev.54.0.flutter-46ab040e58

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).
      If Android SDK has been installed to a custom location, set $ANDROID_HOME to that location.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.3.1, Build version 9E501
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.2

[✗] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).

[✓] VS Code (version 1.23.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Dart Code extension version 2.12.1

[✓] Connected devices (1 available)
    • Seth's iPod touch • 8b28709d2292b357854d7702f39ac581a19624e2 • ios • iOS 9.3.5

! Doctor found issues in 2 categories.
```

**Comment by nokia6290** [comment]
You can modify the flutter file "date_picker.dart" by wrapping _DatePickerDialog with "SingleChildScrollView"
Like in the example:
Widget child = SingleChildScrollView(
child: _DatePickerDialog(
initialDate: initialDate,
firstDate: firstDate,
lastDate: lastDate,
selectableDayPredicate: selectableDayPredicate,
initialDatePickerMode: initialDatePickerMode,
)
);

Side effect: scrolling blue effect is active( haven't figured out how to turn it off ) & once you will try running 'Flutter Upgrade' the file will be overwritten.

**Comment by clemkofi** [comment]
Yes, you have to modify the "date_picker.dart" file but to avoid all the "Side Effects" of using the "SingleChildScrollView" here you can create new const variables for smaller screens and use conditional statements to select them if those conditions (i.e conditions of a "smaller screen" is met) like so

creating the variables at that the start of the class
```dart
// smaller screens parameters (screens with height smaller than 521)
const double _kDatePickerHeaderPortraitHeight_small = 80.0;
const double _kDialogActionBarHeight_small = 45.0;
const double _kDatePickerLandscapeHeight_small = _kMaxDayPickerHeight + _kDialogActionBarHeight_small;
```
then get the screen dimensions for the device at the start of the _DatePickerHeader widget 
```dart
 Widget build(BuildContext context) {
    // get screen dimensions of the phone 
    final ui.Size logicalSize = MediaQuery.of(context).size;
    final double screenWidth = logicalSize.width;
    final double screenHeight = logicalSize.height;

```
now use the device height to set the height in potrait and landscape modes
```dart
switch (orientation) {
      case Orientation.portrait:
        height = screenHeight < 525 ? _kDatePickerHeaderPortraitHeight_small : _kDatePickerHeaderPortraitHeight; //new
        padding = const EdgeInsets.symmetric(horizontal: 16.0);
        mainAxisAlignment = MainAxisAlignment.center;
        break;
      case Orientation.landscape:
        width = screenHeight < 525 ? _kDatePickerLandscapeHeight_small : _kDatePickerHeaderLandscapeWidth; //new
        padding = const EdgeInsets.all(8.0);
        mainAxisAlignment = MainAxisAlignment.start;
        break;
    }
```
This solves the problem for smaller screens. :wink:

**Comment by darrenaustin** [comment]
Closed with #31514
